pkg/management/component_test.go

```diff
@@ -1224,11 +1224,74 @@ func TestResetPassword(t *testing.T) {
 			Value: &password,
 		}
 
-		err := managementComponent.ResetPassword(ctx, "master", userID, passwordRep)
+		_, err := managementComponent.ResetPassword(ctx, "master", userID, passwordRep)
 
 		assert.Nil(t, err)
 	}
 
+	// No password offered
```

Comment:
Finally, I have a last suggestion:
In internal/passwordgeneration.go, you can create an interface PasswordGenerator
CreateGenerator(policy *string) check if policy is nil and creates an instance of type defaultPasswordGenerator or of type keycloakPolicyPasswordGenerator. Both types are implementing the interface (with the functions you already wrote)
During the creation of keycloakPolicyPasswordGenerator, you parse the policy to a map[string]int
You can now easiliy generate a password (and re-generate it if password==username) with the best performances.
And here, in the management testing part, you don't have to worry of the generation, just check that the GeneratePassword function of the mock is called